blues/tests/test_smartdart.py

```diff
@@ -0,0 +1,73 @@
+import unittest, parmed
+from blues.simulation import SimulationFactory
+from blues.moves import SmartDartMove
+from blues.engine import MoveEngine
+from openmmtools import testsystems
+import simtk.unit as unit
+import numpy as np
+
+
+class MoveProposalTester(unittest.TestCase):
+    """
+    Test the MoveProposal class.
+    """
+    def setUp(self):
+        # Obtain topologies/positions
+        testsystem = testsystems.AlanineDipeptideVacuum(constraints=None)
+        structure = parmed.openmm.topsystem.load_topology(topology=testsystem.topology,
+                                            system=testsystem.system,
+                                            xyz=testsystem.positions,
+                                            )
+
+        #self.atom_indices = utils.atomIndexfromTop('LIG', structure.topology)
+        self.functions = { 'lambda_sterics' : 'step(0.199999-lambda) + step(lambda-0.2)*step(0.8-lambda)*abs(lambda-0.5)*1/0.3 + step(lambda-0.800001)',
+                           'lambda_electrostatics' : 'step(0.2-lambda)- 1/0.2*lambda*step(0.2-lambda) + 1/0.2*(lambda-0.8)*step(lambda-0.8)' }
+        self.opt = { 'temperature' : 300.0, 'friction' : 1, 'dt' : 0.002,
+                'nIter' : 10, 'nstepsNC' : 10, 'nstepsMD' : 50,
+                'nonbondedMethod' : 'NoCutoff', 'nonbondedCutoff': 10, 'constraints': 'HBonds',
+                'trajectory_interval' : 10, 'reporter_interval' : 10,
+                'platform' : None,
+                'verbose' : False }
+
+
+        #Initialize the Model object
+        basis_particles = [0,2,7]
+        self.move = SmartDartMove(structure, basis_particles=basis_particles,
+                                resname='ALA')
+        self.move.atom_indices = range(22)
+        self.move.topology = structure.topology
+        self.move.positions = structure.positions
+        self.move.calculateProperties()
+        self.move_engine = MoveEngine(self.move)
+
+        #Initialize the SimulationFactory object
+        sims = SimulationFactory(structure, self.move_engine, **self.opt)
+        system = sims.generateSystem(structure, **self.opt)
+        alch_system = sims.generateAlchSystem(system, self.move.atom_indices)
+        self.nc_sim = sims.generateSimFromStruct(structure, alch_system, ncmc=True, **self.opt)
+        self.move.calculateProperties()
+        self.initial_positions = self.nc_sim.context.getState(getPositions=True).getPositions(asNumpy=True)
+
+    def test_smart_dart(self):
```

Comment:
Can you just, er, SAY what the test is? I don't have much time to digest it, but it looks like you're taking alanine dipeptide and ... ummm... ? And if you just have alanine dipeptide, what role do the basis particles serve? 

I'm having trouble figuring out what is being moved in this test, and what it's being moved relative to. And, how do we know if it's correct or is this just making sure it works?